dipy/nn/deepn4.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/python
+"""
+Class and helper functions for fitting the DeepN4 model.
+"""
+import logging
+import numpy as np
+
+import nibabel as nib
+from scipy.ndimage import gaussian_filter
+from dipy.nn.utils import transform_img, recover_img
+
+from dipy.data import get_fnames
+from dipy.testing.decorators import doctest_skip_parser
+from dipy.utils.optpkg import optional_package
+from dipy.nn.utils import set_logger_level
+
+tf, have_tf, _ = optional_package('tensorflow')
+tfa, have_tfa, _ = optional_package('tensorflow_addons')
+if have_tf and have_tfa:
+    from tensorflow.keras.models import Model
+    from tensorflow.keras.layers import MaxPool3D, Conv3DTranspose
+    from tensorflow.keras.layers import Conv3D, LeakyReLU
+    from tensorflow.keras.layers import Concatenate, Layer
+    from tensorflow_addons.layers import InstanceNormalization
+else:
+    logging.warning('This model requires Tensorflow.\
```

Comment:
Also mention Tensorflow add-ons, so like the Synb0 message

Reply:
updated to 
```
logging.warning('This model requires Tensorflow and Tensorflow\
                    -addons. Please install these packages using \
                    pip. If using mac, please refer to this \
                    link for installation. \
                    https://github.com/apple/tensorflow_macos')
```